tests/test_projection_schema_migrations.py

```
@@ -0,0 +1,360 @@
+    db_path = vault / "60-Logs" / "knowledge.db"
+    _make_v7_db(db_path)
+
+    monkeypatch.setattr(ki, "_ensure_authority_schema_version", lambda _vault: 1)
+    monkeypatch.setattr(ki, "_knowledge_db_supports_pack_schema", lambda _db_path: True)
+    monkeypatch.setattr(
+        ki,
+        "rebuild_knowledge_index",
+        lambda *_a, **_kw: pytest_fail("delta path should fail, not full-rebuild fallback"),
+    )
+
+    def _boom(*_args, **_kwargs):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_boom`

Add return type annotation: `NoReturn`

(ANN202)